src/i2c_devices.rs

```diff
@@ -0,0 +1,104 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use cc3200::I2C;
+
+// This implementation is a port of example/out_of_box/tmp006drv.c
+
+static TMP006_DEV_ADDR: u8 = 0x41;
+static TMP006_VOBJECT_REG_ADDR: u8 = 0x00;
+static TMP006_TAMBIENT_REG_ADDR: u8 = 0x01;
+static TMP006_CONFIG_REG_ADDR: u8 = 0x02;
+static TMP006_MANUFAC_ID_REG_ADDR: u8 = 0xFE;
+static TMP006_DEVICE_ID_REG_ADDR: u8 = 0xFF;
+
+static TMP006_MANUFAC_ID: u16 = 0x5449;
+static TMP006_DEVICE_ID: u16 = 0x0067;
+
+pub struct Temperature {
```

Comment:
Should this be called I2C_Device (i.e. a device sitting on an I2C Bus)?

At the very least it should probably be Sensor specific (i.e. TMP006). Because the code here probably won't work with the HDC1010